content/ADR-164-quests.poc.md

```diff
@@ -0,0 +1,211 @@
+---
+layout: adr
+adr: 164 
+title: Quests PoC 
+date: 2022-12-28
+status: Draft 
+type: RFC 
+spdx-license: CC0-1.0
+authors:
+  - guidota 
+---
+
+## Abstract
+
+The Quests System is an important feature that facilitates users to explore the world, unlock achievements and potentially receive rewards. A quest is a series of steps or tasks that a user has to complete. Each step or task has an acceptance criteria to consider it as done. A quest designer has to define the steps and the order or the path to the end, so the quest is finished when those steps are completed.
+
+## Context, Reach & Prioritization
+
+Users use to complain that when they enter the world and after a certain time, they don't know what to do, or where to go and feel lost. This problem impacts directly the retention of the users and the time quality of the first steps in the platform.
+
+Having a Quests System can solve part of the problem, giving users a reason to stay and have a nice time. This feature would also allow content creators to be creative and increase engagement.
+
+The last sentence is important to remark on since it is one of the differences to any other Quest System you may know from video games, where the quests are defined by the owners or internal teams. In this case, there should be a focus on creating easy tools for quest designers.
+
+On the other side, the system (backend) should be in charge of tracking and validating the progress and accepting new quest deployments.
+
+Having said that, there are three well-known actors with corresponding components:
+ - Users: Explorer and Kernel 
+ - Designers: Visual tool and/or CLI
+ - The System: Backend
+
+For the sake of fast iteration and discovery, this document will explore the boundaries and scope of a Proof of Concept for the described Quests System.
+
+## Solution Space Exploration
+
+The following sections will describe what would be necessary and alternatives for this Proof of Concept, where designers can submit quests, users experience them and the system expose some metrics. 
+
+### Quest definition
+
+A Quest definition will include steps/tasks required for completion. It would be defined technically speaking as a graph, since it may have a non-linear complexity.
+
+Defining a valid graph, as it may require some conditions:
+- it should have only one start node.
+- it should have at least one end node.  
+- it may have cycles but not endless loops.
+- it may have conditional nodes.
+
+### Explorer
+
+Minimal UI for quest tracking and receive real-time notifications about the progress:
```

Comment:
The UI for quest tracking should also be global, making it easy for users to show/hide whenever they want.
```suggestion
Minimal global UI for quest tracking and receive real-time notifications about the progress:
```